From d0861ceacbe61c3325526eacfddf8231286b4864 Mon Sep 17 00:00:00 2001
From: Michal Sekletar <msekleta@redhat.com>
Date: Thu, 13 Dec 2012 14:59:39 +0100
Subject: [PATCH 160/204] path-util: set pointer to null after calling free()

In cases where path_strv_canonicalize() returns NULL, strv_free() is
called afterwards and it will call free() on pointers which were freed
already in path_strv_canonicalize()
(cherry picked from commit c9c7aef24f9750bbd51d2c13beff66473d96bf3c)
---
 src/shared/path-util.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/shared/path-util.c b/src/shared/path-util.c
index 70c8a8a..dd12d3d 100644
--- a/src/shared/path-util.c
+++ b/src/shared/path-util.c
@@ -181,6 +181,7 @@ char **path_strv_canonicalize(char **l) {
 
                 t = path_make_absolute_cwd(*s);
                 free(*s);
+                *s = NULL;
 
                 if (!t) {
                         enomem = true;
-- 
1.8.1.5

